### ISSUE TITLE
Panics result in empty responses unless they are intercepted in middleware

### ISSUE THREAD

**Issue by marioizquierdo** [twitchtv/twirp#162]
Auto-generated code for Golang Twirp services seem to be handling panics:
https://github.com/twitchtv/twirp/blob/311d20ef3700822f5683641c9136d85fd5e14422/protoc-gen-twirp/generator.go#L1020

### Expected behavior

The code suggests that panics should result in a `twirp.Internal` error like this:
```go
# curl -X POST http://localhost:9090/twirp/mypkg.HelloWorld/Hello ....
{"code": "internal", "msg": "Internal service panic"}
```

### Actual behavior

However, they result in an empty response. Doing a curl request looks like this:
```sh
# curl -X POST http://localhost:9090/twirp/mypkg.HelloWorld/Hello ....
curl: (52) Empty reply from server
```

### Intercepting the panic in middleware

Adding middleware to intercept the panic allows the response to properly flush:

```go
func InterceptPanics(h http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		defer func() {
			if p := recover(); p != nil {
				// do whatever, but don't re-panic
			}
		}()
		h.ServeHTTP(w, r)
	})
}
```

### Using w.(http.Flusher).Flush()

Maybe Twirp should make sure that the internal error response should be flushed before re-throwing the panic?

I tried changing the auto-generated code, adding `w.(http.Flusher).Flush()` right before re-throwing the panic and it generates a valid response, however the http connection was in a chunked mode and curl was awaiting for more chunks or 0x00 0x00 sequence. Instead it got a connection closed.

```
# curl -X POST http://localhost:9090/twirp/mypkg.HelloWorld/Hello ....
curl: (18) transfer closed with outstanding read data remaining
{"code": "internal", "msg": "Internal service panic"}
```

**Comment by spenczar** [comment]
I think we should Flush in the generated code.

We should be able to explicitly set `Content-Length`, which avoids switching us into chunked transfer encoding, as documented in [http.ResponseWriter.Write](https://godoc.org/net/http#ResponseWriter).

**Comment by spenczar** [comment]
Fixed in #168, release v5.7.0.
